src/KilnBase.sol

```diff
@@ -57,67 +48,75 @@ abstract contract DssKiln {
     }
 
     modifier auth {
-        require(wards[msg.sender] == 1, "DssKiln/not-authorized");
+        require(wards[msg.sender] == 1, "KilnBase/not-authorized");
         _;
     }
 
-    // --- Mutex  ---
     modifier lock {
-        require(locked == 0, "DssKiln/system-locked");
+        require(locked == 0, "KilnBase/system-locked");
         locked = 1;
         _;
         locked = 0;
     }
 
-    function _min(uint256 x, uint256 y) internal pure returns (uint z) {
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
         return x <= y ? x : y;
     }
 
+    /**
+        @dev Auth'ed function to authorize an address for privileged functions
+        @param usr   Address to be authorized
+    */
     function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
 
+    /**
+        @dev Auth'ed function to un-authorize an address for privileged functions
+        @param usr   Address to be un-authorized
+    */
     function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
 
     /**
         @dev Auth'ed function to update lot or hop values
         @param what   Tag of value to update
         @param data   Value to update
     */
-    function file(bytes32 what, uint256 data) external auth {
+    function file(bytes32 what, uint256 data) public virtual auth {
         if      (what == "lot") lot = data;
         else if (what == "hop") hop = data;
-        else revert("DssKiln/file-unrecognized-param");
+        else revert("KilnBase/file-unrecognized-param");
         emit File(what, data);
     }
 
     /**
         @dev Auth'ed function to withdraw unspent funds
+        @param dst   Destination of the funds
     */
-    function rug() external auth {
+    function rug(address dst) external auth {
         uint256 amt = GemLike(sell).balanceOf(address(this));
-        GemLike(sell).transfer(msg.sender, amt);
-        emit Rug(msg.sender, amt);
+        GemLike(sell).transfer(dst, amt);
+        emit Rug(dst, amt);
     }
 
     /**
         @dev Function to execute swap/drop and reset zzz if enough time has passed
     */
     function fire() external lock {
-        require(block.timestamp >= zzz + hop , "DssKiln/fired-too-soon");
-        uint256 _amt = _min(GemLike(sell).balanceOf(address(this)), lot);
-        require(_amt > 0, "DssKiln/no-balance");
-        uint256 _swapped = _swap(_amt);
+        require(block.timestamp >= zzz + hop, "KilnBase/fired-too-soon");
+        uint256 amt = _min(GemLike(sell).balanceOf(address(this)), lot);
+        require(amt > 0, "KilnBase/no-balance");
+        uint256 swapped = _swap(amt);
         zzz = block.timestamp;
```

Comment:
Mmm what's the motivation for storing the next fireable time instead of the last fire time?

With the way it is now I think updating the value is more straightforward (for example if you want a shorter/longer `hop` it will just take effect without needing to calculate anything special).
With the suggested change, in order to change `hop` and not add any time delay you would need to complicate `file()` with something like:
https://github.com/makerdao/curve-lp-oracle/blob/master/src/CurveLPOracle.sol#L150

Reply:
Well the complication of `file` is for the new `hop` to take effect immediately after it is filed, which is at least the behavior in CurveLpOracle (so no aiming for that kind of keeper predictability there, and at least the keepers developed lately check `fire`ability/`execute`ability each block).

Note that being able to change `hop` immediately allows you to set a very long `hop` and then reduce it if you want (without waiting forever to reach `zzz`). So I thing the ability to call `fire` sooner is a feature, not a bug (and at least is regarded that way in the oracles).

Reply:
ok, I didn't see the complication because the benefit I saw to my suggestion was that governance couldn't make the new `hop` take effect immediately (if that is the goal,I agree your approach is the way to go).

I think I disagree and would say we want to prevent governance from having the ability to immediately mess with the fire times, but I don't feel strongly enough to block on this.  Lets leave the comments open and see if others weigh in and have any thoughts